invoices/invoiceregistry.go

```diff
@@ -1254,9 +1263,16 @@ func (i *InvoiceRegistry) SubscribeNotifications(addIndex, settleIndex uint64) *
 	select {
 	case i.newSubscriptions <- client:
 	case <-i.quit:
+		return nil, ErrShuttingDown
 	}
 
-	return client
+	select {
+	case <-client.processed:
```

Comment:
Copy explanation in pr desc to commit message body?

Reply:
I don't fully understand the problem yet. We let the backlog pile up in the queue before returning this call, but what would that fix?

Reply:
sorry, ignore the original commit body on this one (updated now tho). 

originally this was meant following concurrency issue when there is a single, single-threaded client
```golang
    // wants full backlog, db has invoices 1-5
    client := i.SubscribeNotifications(1, 1)
    
    // invoice6 is added to database before InvoicesAddedSince(1) is called
    i.AddInvoice(invoice6)

    for event := range client.NewUpdates {
        // dulicate invoice 6
    }
```

however, the final commit actually addresses this in a stronger sense since it guarantees that `InvoicesAddedSince` and `InvoicesSettledSince` are executed atomically, which fixes the issue in the multi-client case.

now this commit just ensure that we don't return a dead client when `quit` is signaled above